raiden-ts/src/transfers/epics.ts

```diff
@@ -106,26 +106,42 @@ function makeAndSignTransfer(
         return EMPTY;
       }
 
-      // assume metadata is valid and recipient is first hop of first route
-      const metadata: Metadata = action.payload.metadata,
-        recipient = metadata.routes[0].route[0];
+      // assume paths are valid and recipient is first hop of first route
+      // compose metadata from it, and use first path fee
+      const metadata: Metadata = {
+          routes: action.payload.paths.paths.map(({ path }) => ({ route: path })),
+        },
+        fee = action.payload.paths.paths[0].fee,
```

Comment:
The `action.payload.paths.paths` being used 3 times here looks rather cumbersome, e.g. why the doubled `paths.paths`?

Reply:
The second `paths` is called `result` in the `PathResults`, but I think `paths` is more descriptive since it's an array of objects whose main property is `path`, which is the path as an array of addresses. The outer `paths` is just because we're refering to the passed `Paths`. I.e.: second paths is the array, first paths is the `Paths` object containing all routing information